### PR TITLE
feat(llm): add ChatContext 

### PR DESCRIPTION
**Pull request by nbsp**
incomplete implementation, notably missing tool call hooks (see comment
for more info).

### CODE REVIEW

File: agents/src/llm/chat_context.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  ASSISTANT,
+  TOOL,
+}
+
+export interface ChatImage {
+  image: string | AudioFrame;
+  inferenceWidth?: number;
+  inferenceHeight?: number;
+  /** Used by LLM implementations to store a processed version of the image for later use. */
+  cache: { [id: string | number | symbol]: any };
```
Comment (theomonnom) [inline agents/src/llm/chat_context.ts:22]: any way to make sure "cache" is private?

Comment (nbsp) [reply agents/src/llm/chat_context.ts:22]: to what degree? LLM needs direct access to it, so we can't make it private or readonly. i can mark is as `@internal` so it's clear only llms should use it

Comment (theomonnom) [reply agents/src/llm/chat_context.ts:22]: internal sounds good

**Review by theomonnom — APPROVED** [review]

File: plugins/openai/src/realtime/realtime_model.ts [hunk 2]
```diff
@@ -129,12 +138,127 @@ class ConversationItem {
...
   }
 
-  create(item: api_proto.ConversationItemCreateContent, previousItemId?: string): void {
-    this.#session.queueMsg({
-      type: 'conversation.item.create',
-      item,
-      previous_item_id: previousItemId,
-    });
+  // create(item: api_proto.ConversationItemCreateContent, previousItemId?: string): void {
+  create(message: llm.ChatMessage, previousItemId?: string): void {
```
Comment (theomonnom) [inline plugins/openai/src/realtime/realtime_model.ts:142]: 🔥

**Pull request merged**